### PR TITLE
ISIC 2018 Improved UNET architecture

### PR DESCRIPTION
**Pull request by isak-ow**
This pullrequest contains my attemt at implementing the improved unet model proposed by the paper in the assignment. Because of the amount of jobs in queue for gpu on the rangpur cluster, I had to train the model in google colab (same code). After training the model it achieved a dice coefficient of 0.8 and an accuracy of 0.9 (may be a wierd metric). See the README.md for more information.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Well organized files, but wrong files path. The student constructed document should be packed in the ‘recognition’ document.
- Algorithm can solve the problem, but no model training evidence/record. The err.txt shows some training error occurred, and no out.txt content, both of which indicates a failed training attempt. 
- Try to use PyTorch to construct the Improved U-Net model, and the model body structure seems good. Some useless functions in utilities.py file. No data augmentation.
- Insufficient comment in the code, no docstring, also need some references.
- Easy difficulty problem, no higher difficulty marks.

**Comment by isak-ow** [comment]
I changed the file path according to the feedback from Zhao, and added a training script from Colab. As mentioned in the Readme, the training I did had to be done on colab due to the queue on the rangpur cluster.

**Comment by shakes76** [comment]
# Marking
Marked based on content submitted before due date.
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, but shows error message -3
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem, no model training evidence -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
No Data leakage, no augmentation -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages minimal -1
Progressive commits used
## Documentation
ReadMe OK, no usage -1, no refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in InCorrect Branch) initial PR -2
Feedback required, log directories and pyc files need to be removed -2
Request Description good

**Comment by shakes76** [comment]
Will not merge due to log files commit and would make repo too large. Feedback marks possible +2 if the requested changes are made (see above), but no merge will be done.

**Comment by isak-ow** [comment]
Hi,

In the initial inspection of the code I was told that some changes could be
made, but that no action was required for the project to pass. After
reading your grading it has me worried that I will not pass the course. I
will of course address the issues I can, raised in your comment, but Im
quite stressed that I will recieve a failing grade. Could you speak to this?

Kind regards,

Isak Orlando Wangensteen
studentID: 48201292

**Comment by isak-ow** [comment]
Removed the log directories and the pyc files as requested in the feedback

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
